### ISSUE TITLE
Reason for 999ms delay in onload?

### ISSUE THREAD

**Issue by agorilla** [aFarkas/lazysizes#517]
Hi, 

I'm sure there is a reason for the the 999ms delay in the onload method but can you tell my why it's necessary?

https://github.com/aFarkas/lazysizes/blob/e305d28d4b6c4ff9f12dd4d438d5d7c5d41574db/lazysizes.js#L540-L543

We use lazysizes but it takes the extra 999ms to show the images within the viewport with the default lazysizes config.

 I'd like to understand why the delay was added because currently we're looking for a workaround which involves some hacks to speed up the image loading.

Thanks
Simon

**Comment by aFarkas** [comment]
Images inside the viewport are loaded immediately. The one second has to do with preloading images that are not in view. You can easily change this by changing `loadMode` to 3.

**Comment by agorilla** [comment]
Thanks for the explanation.

The images are inside the viewport and loadMode change did not solve the delay problem.

But with further analysis I found out that we hide the lazyloading `<img>` with `visibility: hidden` and not as in the [documentation recommended](https://github.com/aFarkas/lazysizes#css-api) with `opacity: 0`.

After switching to `opacity: 0` lazysizes works perfectly without the delay.

**tl;dr:** Our CSS implementation caused the delay so I close this issue.
